src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt

```diff
@@ -134,3 +134,10 @@ implication        ,ARM64 ,Sha1      ,ArmBase
 implication        ,ARM64 ,Sha256    ,ArmBase
 implication        ,ARM64 ,Vector64  ,AdvSimd
 implication        ,ARM64 ,Vector128 ,AdvSimd
+
+; Known CPUs (instruction groups)         ,lower baselines included by implication
+instructionsetgroup  ,sandybridge  ,X64   ,avx pclmul popcnt
+instructionsetgroup  ,sandybridge  ,X86   ,avx pclmul popcnt
+instructionsetgroup  ,haswell      ,X64   ,avx2 bmi fma lzcnt pclmul popcnt movbe
+instructionsetgroup  ,haswell      ,X86   ,avx2 bmi fma lzcnt pclmul popcnt movbe
```

Comment:
If we expose anything here, I don't see why `sandybridge` and `haswell` are the ones we're exposing.

`Skylake` is the version we base our perscore numbers and many of the other metrics on. `Cascade Lake` is the one we use in many modern benches.

Or even better, there were 4 common x86 ISA version officially defined back in 2020. This was a joint effort among AMD, Intel, Red Hat, SUSE, and of course the Clang and GCC compilers: https://lists.llvm.org/pipermail/llvm-dev/2020-July/143289.html

These common ISA versions are:
* `x86-x64`: `cmov`, `cx8`, `fpu`, `fxsr`, `mmx`, `osfxsr`, `sce`, `sse`, `sse2`
* `x86-x64-v2`: `...`, `cx16b`, `lahf-sahf`, `popcnt`, `sse3`, `sse4.1`, `sse4.2`, `ssse3`
* `x86-x64-v3`: `...`, `avx`, `avx2`, `bmi1`, `bmi2`, `f16c`, `fma`, `lzcnt`, `movbe`, `osxsave`
* `x86-x64-v4`: `...`, `avx512f`. `avx512bw`, `avx512cd`. `avx512dq`, `avx512l`

where `...` means "everything in the previous version
where `sce` means "syscall", `cx8/16` are "compare exchange", `fpu` is the x87 FPU

where x86-v2 is ~2009 (nehalem for Intel/jaguar for AMD)
where x86-v3 is ~2015 (haswell for Intel/excavator for AMD)
where x86-v4 doesn't have a common year yet.

Reply:
These common versions are likely the "best" and most agnostic thing if we're only supporting a minimal subset and not trying for parity with Clang/GCC.

They represent an actual "standardized agreement" (for lack of a better term) on what these baselines are from the actual CPU manufacturers and so are the closest to having a well-defined standard here.

Reply:
Arm has similar version scheme: ARMv8, ARMv9. If we go with the versions for x64, we can do the same for Arm.

Reply:
added, can you review please? I asseum we don't need to add armv9 since we have nothing to offer for it

Another question is why we don't set `sha1` and `sha2` for the baseline on arm64? it seems to be part of 8.0-a isn't it?